File: backend-rust/src/domain/models/session.rs
use chrono::NaiveDateTime;
use serde::Serialize;
use uuid::Uuid;

#[derive(Serialize)]
pub struct Session {
    pub id: String,
    pub user_id: Option<Uuid>,
    pub csrf_token: String,
    pub ip_address: Option<String>,
    pub user_agent: Option<String>,
    pub expired_at: NaiveDateTime,
    pub created_at: NaiveDateTime,
    pub updated_at: NaiveDateTime,
}

#[derive(Serialize)]
pub struct NewSession {
    pub id: String,
    pub user_id: Option<Uuid>,
    pub csrf_token: String,
    pub expired_at: NaiveDateTime,
}

File: backend-dotnet/src/Backend.Domain/Entities/Session.cs
namespace Backend.Domain.Entities;

public class Session
{
    public required string Id { get; set; }
    public Guid? UserId { get; set; }
    public required string CsrfToken { get; set; }
    public string? IpAddress { get; set; }
    public string? UserAgent { get; set; }
    public DateTime ExpiredAt { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.Now;
    public DateTime UpdatedAt { get; set; } = DateTime.Now;

    public User? User { get; set; } = null!;
}
